app/src/main/java/com/mkulesh/onpc/rememberedgroups/GroupMembership.java

```diff
@@ -0,0 +1,36 @@
+package com.mkulesh.onpc.rememberedgroups;
+
+import androidx.annotation.NonNull;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+@Entity(primaryKeys = {"group_name", "device"})
+public class GroupMembership {
+
+    // TODO the corresponding master device must be saved too
+    // Blocker: I have not yet figuered out how the master device is represented in the
+    // addCmd etc commands
+    // Also, some general understanding of the commumication with the speakers
+    // would be helpful, so that we can store all information that is needed to
+    // enable / disable stored groups from android 8+ homescreen shortcuts
+    // *without* opening the corresponding view in the app
```

Comment:
The master device is the device to which the app is currently connected. Device that receives MultiroomGroupSettingMsg (MGS) message with target group configuration and process it, becomes a master status. See MultiroomManager::sendGroupUpdates for more details (this method builds the command to create/remove the group and always sends this command to one currently connected device represented by activity.getStateManager()). 

In the communication with the speakers, there are three major cases:
- There is no group yet. You send MGS with purposes "Create a group" to the currently connected device. After the group is created, this device becomes a master.
- There is already a group. You open "Group/Ungroup devices" dialog for any slave device, change group configuration it and sends new MGS with purposes "Create a group" to this slave device. After the group is updated, this device becomes a master.
- There is already a group. You open "Group/Ungroup devices" dialog for any device (master or any slave) and sends MGS with purposes "Delete group". Group shall be deleted, there are no more master or slave devices.

It is the reason why the currently connected device in the "Group/Ungroup devices" dialog is disabled for slave selection since it will be handled as a master. In order to make a dedicated device as a master, the user shall fist connect to it from the app and than configure a group for it.

In order to access the parameters of master device, please pass the Configuration object here and use two getters of the Configuration class: getDeviceName and getDevicePort. This pair (getDeviceName/getDevicePort) shall be stored in your DB.    

In order to restore stored configuration, please access activity.getDeviceList().getDevices(), search there the device that has the stored name/port (like in method MonitorFragment::updateMultiroomGroupBtn) and just send the MGS to this device. 

If you have any questions, please contact me per email: mikhail.kulesh@gmail.com